qemu/tests/x86_cpu_L3_cache.py

```diff
@@ -31,6 +31,8 @@ def boot_and_check_guest(machine_type, check_L3=False):
         params['machine_type'] = machine_type
         params['start_vm'] = 'yes'
         vm_name = params['main_vm']
+        if params.get_numeric('vcpu_maxcpus') > 128:
```

Comment:
1. Where is the machine type limitation?
2. Should also modify `smp` together.

Reply:
> 1. Where is the machine type limitation?
Machine type 'pc-q35-rhel7.3.0 ' which doesn't support L3 to be desgined in test case step. It used to be compared with the newest machine type.

> 2. Should also modify `smp` together.
What's this meaning?

Reply:
`smp` and `vcpu_maxcpus` can both set the CPU topology, if the case does not include CPU hotplug, better to let them with the same value. So here I think should check `if max(vcpu_maxcpus, smp) > 128`, then `smp == vcpu_maxcpus = 128`

Reply:
> `smp` and `vcpu_maxcpus` can both set the CPU topology, if the case does not include CPU hotplug, better to let them with the same value. So here I think should check `if max(vcpu_maxcpus, smp) > 128`, then `smp == vcpu_maxcpus = 128`

Thanks, Update and test pass on 143 cpus host with '--customsparams="vcpu_maxcpus = 140"'.

 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.x86_cpu_L3_cache.q35: STARTED
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.x86_cpu_L3_cache.q35: PASS (80.86 s)